### PR TITLE
Replace NodesInfoMetrics.Metric.allMetrics() with immutable ALL_METRICS

### PR DESCRIPTION
**Pull request by sabi0**
The method was unnecessarily building a new Set on every call.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequest.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public Set<String> requestedMetrics() {
      * Add metric
      */
     public NodesInfoRequest addMetric(String metric) {
-        if (NodesInfoMetrics.Metric.allMetrics().contains(metric) == false) {
+        if (NodesInfoMetrics.Metric.ALL_METRICS.contains(metric) == false) {
             throw new IllegalStateException("Used an illegal metric: " + metric);
```
Comment (sabi0) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequest.java:76]: Probably `IllegalArgumentException` would be a more suitable option here?

**Comment by astefan** [comment]
buildkite test this

**Comment by dakrone** [comment]
@elasticmachine test this

**Comment by dakrone** [comment]
@sabi0 can you add a changelog to `/docs/changelog/105283.yaml` for this PR? It's normally added by a bot but it appears not to have been added this time. You can look at the other changelogs in the same folder to see the format.

**Comment by sabi0** [comment]
Done.
Please let me know if some other "area" should be used instead of "Data streams".

**Comment by sabi0** [comment]
Is there anything that still needs to be done here?

**Comment by dakrone** [comment]
@elasticmachine update branch

**Comment by dakrone** [comment]
@elasticmachine test this

**Review by dakrone — APPROVED** [review]
LGTM

**Comment by dakrone** [comment]
@elasticmachine update branch

**Comment by dakrone** [comment]
@elasticmachine test this

**Comment by dakrone** [comment]
Sorry, but I've been traveling for the last few days, and it looks like this has been closed by #105888 in the meantime, so I'm going to close this. Thanks!

**Pull request closed without merging**